File: app/models/statistics.py
"""Response models for dataset statistics endpoint."""

from pydantic import BaseModel


class ClassDistribution(BaseModel):
    """Per-category annotation counts split by source (GT vs prediction)."""

    category_name: str
    gt_count: int
    pred_count: int


class SplitBreakdown(BaseModel):
    """Sample count per dataset split (train/val/test/unassigned)."""

    split_name: str
    count: int


class SummaryStats(BaseModel):
    """Aggregate counts across the entire dataset."""

    total_images: int
    gt_annotations: int
    pred_annotations: int
    total_categories: int


class DatasetStatistics(BaseModel):
    """Combined statistics payload for a single dataset."""

    class_distribution: list[ClassDistribution]
    split_breakdown: list[SplitBreakdown]
    summary: SummaryStats

File: frontend/src/types/statistics.ts
/**
 * Types for the dataset statistics API response.
 *
 * Mirrors the backend models in app/models/statistics.py.
 */

export interface ClassDistribution {
  category_name: string;
  gt_count: number;
  pred_count: number;
}

export interface SplitBreakdown {
  split_name: string;
  count: number;
}

export interface SummaryStats {
  total_images: number;
  gt_annotations: number;
  pred_annotations: number;
  total_categories: number;
}

export interface DatasetStatistics {
  class_distribution: ClassDistribution[];
  split_breakdown: SplitBreakdown[];
  summary: SummaryStats;
}
